File: Python/AD-HOC/2286 - Par ou Ímpar.py
"""
Autor: Pierre Vieira
Data da submissão: 23/02/2020 12:01:14
"""


def solve_this_problem():
    nome_par, nome_impar = input(), input()
    for c in range(n):
        n1, n2 = map(int, input().split())
        if (n1 + n2) % 2 == 0:
            print(nome_par)
        else:
            print(nome_impar)


cont = 1
while True:
    n = int(input())
    if n == 0:
        break
    print('Teste {}'.format(cont))
    solve_this_problem()
    print()
    cont += 1

File: Kotlin/AD - HOC/2286 - Par ou Ímpar.kt
/*
* Autor: Pierre Vieira
* Data de submissão: 21/03/2023 00:24:18
*/

fun main(args: Array<String>) {
    var turn = 1
    while (true) {
        val numberOfTurns = readLine()!!.toInt()
        if (numberOfTurns == 0) break
        println("Teste $turn")
        solution(numberOfTurns)
        println()
        turn++
    }
}

private fun solution(numberOfTurns: Int) {
    val firstName = readLine()!!
    val secondName = readLine()!!
    repeat(numberOfTurns) {
        val sum = readLine()!!.split(" ").map { it.toInt() }.sum()
        println(if (sum.isEven()) firstName else secondName)
    }
}

private fun Int.isEven(): Boolean = this % 2 == 0